akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -233,16 +233,19 @@ object Source extends SourceApply {
   /**
    * Create a `Source` that will continually emit the given element.
    */
-  def repeat[T](element: T): Source[T, Unit] =
+  def repeat[T](element: T): Source[T, Unit] = {
+    ReactiveStreamsCompliance.requireNonNullElement(element)
```

Comment:
I don't think this should be checked here. Is there any reason to believe that a null element in here would be different from pulling out a null from an Iterable?